core/src/main/scala/org/apache/spark/shuffle/sort/SortShuffleWriter.scala

```diff
@@ -72,6 +72,18 @@ private[spark] class SortShuffleWriter[K, V, C](
       val partitionLengths = sorter.writePartitionedFile(blockId, tmp)
       shuffleBlockResolver.writeIndexFileAndCommit(dep.shuffleId, mapId, partitionLengths, tmp)
       mapStatus = MapStatus(blockManager.shuffleServerId, partitionLengths)
+      partitionLengths.foreach(writeMetrics.incBlockSizeDistribution(_))
+      if (mapStatus.isInstanceOf[HighlyCompressedMapStatus]) {
+        writeMetrics.setAverageBlockSize(
+          mapStatus.asInstanceOf[HighlyCompressedMapStatus].getAvgSize());
+        (0 until partitionLengths.length).foreach {
+          case i =>
+            if (partitionLengths(i) < mapStatus.getSizeForBlock(i)) {
```

Comment:
don't you want the condition reversed? `partitionLengths(i)` is the true size, so an underestimate is if the mapStatus says the size is smaller.